vortex-array/src/arrays/bool/compute/is_constant.rs

```diff
@@ -0,0 +1,13 @@
+use vortex_error::VortexResult;
+
+use crate::arrays::{BoolArray, BoolEncoding};
+use crate::compute::IsConstantFn;
+use crate::Array;
+
+impl IsConstantFn<&BoolArray> for BoolEncoding {
+    fn is_constant(&self, array: &BoolArray) -> VortexResult<Option<bool>> {
+        let true_count = array.boolean_buffer().count_set_bits();
+
+        Ok(Some((true_count == 0) | (true_count == array.len())))
```

Comment:
I think you want to get the first value, then iterate the bool buffer as u64s and check for == 0 or == u64::MAX, exiting early if any are false.